### ISSUE TITLE
json: cannot unmarshal string into Go value of type spec.SchemaProps

### ISSUE THREAD

**Issue by hagh** [mohsen1/multi-file-swagger-example#9]
When I am trying to run swagger on this example I get the following error
```
json: cannot unmarshal string into Go value of type spec.SchemaProps
```
Both "validate" and "generate server" commands immediately return with the same above error.
I tried to make changes to the index file, going step by step (for example by replacing the info part with the corresponding external #ref). Then I can bypass this error but still it does not seem any of those #refs are properly resolved. I am using [go-swagger](https://github.com/go-swagger/go-swagger) to validate and generate the server.

**Comment by mohsen1** [comment]
It's possibly a bug in go-swagger. Have you asked them?

**Comment by hagh** [comment]
No, but we found that this issue https://github.com/go-swagger/go-swagger/issues/358 has been filed with them about two weeks ago, which looks like the same problem that I am complaining about. Can we possibly use the json-refs to resolve the #refs first and then pass the output to go-swagger? thanks.

**Comment by mohsen1** [comment]
That should work as long as you don't have critical references
